### PR TITLE
Force value rendering and validation check after the clear button click (T818673)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by DokaRus — COMMENTED** [review]
lgtm

**Review by DokaRus — DISMISSED** [review]

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]
There are no tests

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 1]
```diff
@@ -585,12 +586,35 @@ QUnit.module("basics", {}, () => {
...
+            value: null
+        });
+        const instance = $element.dxNumberBox("instance");
+        const $input = $element.find(".dx-texteditor-input");
+        const kb = keyboardMock($input);
+
+        kb.type("11");
+
+        const origInputIsInvalid = instance._inputIsInvalid;
+        instance._inputIsInvalid = () => true;
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]:
```js
sinon.stub(instance, "_inputIsInvalid ", () => true);
<...>
instance._inputIsInvalid.restore();
```

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 2]
```diff
@@ -585,12 +586,35 @@ QUnit.module("basics", {}, () => {
...
         assert.strictEqual($input.val(), "", "value is still cleared");
     });
 
+    QUnit.test("clearButton should clear the text and reset incorrect value (T818673)", (assert) => {
+        const $element = $("#numberbox").dxNumberBox({
+            showClearButton: true,
+            value: null
+        });
+        const instance = $element.dxNumberBox("instance");
+        const $input = $element.find(".dx-texteditor-input");
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]:
```suggestion
        const $input = $element.find(`.${INPUT_CLASS}`);
```

**Review by DokaRus — APPROVED** [review]

**Review by mrjohn0011 — APPROVED** [review]

**Pull request merged**